cleverhans_tutorials/tutorial_models.py

```diff
@@ -7,183 +7,42 @@
-
-
-class Conv2D(Layer):
-
-    def __init__(self, output_channels, kernel_shape, strides, padding):
-        self.__dict__.update(locals())
-        del self.self
-
-    def set_input_shape(self, input_shape):
-        batch_size, rows, cols, input_channels = input_shape
-        kernel_shape = tuple(self.kernel_shape) + (input_channels,
-                                                   self.output_channels)
-        assert len(kernel_shape) == 4
-        assert all(isinstance(e, int) for e in kernel_shape), kernel_shape
-        init = tf.random_normal(kernel_shape, dtype=tf.float32)
-        init = init / tf.sqrt(1e-7 + tf.reduce_sum(tf.square(init),
-                                                   axis=(0, 1, 2)))
-        self.kernels = tf.Variable(init)
-        self.b = tf.Variable(
-            np.zeros((self.output_channels,)).astype('float32'))
-        input_shape = list(input_shape)
-        input_shape[0] = 1
-        dummy_batch = tf.zeros(input_shape)
-        dummy_output = self.fprop(dummy_batch)
-        output_shape = [int(e) for e in dummy_output.get_shape()]
-        output_shape[0] = batch_size
-        self.output_shape = tuple(output_shape)
-
-    def fprop(self, x):
-        return tf.nn.conv2d(x, self.kernels, (1,) + tuple(self.strides) + (1,),
-                            self.padding) + self.b
-
-    def get_params(self):
-        return [self.kernels, self.b]
-
-
-class ReLU(Layer):
-
-    def __init__(self):
-        pass
-
-    def set_input_shape(self, shape):
-        self.input_shape = shape
-        self.output_shape = shape
-
-    def fprop(self, x):
-        return tf.nn.relu(x)
-
-    def get_params(self):
-        return []
-
-
-class Softmax(Layer):
-
-    def __init__(self):
-        pass
-
-    def set_input_shape(self, shape):
-        self.input_shape = shape
-        self.output_shape = shape
-
-    def fprop(self, x):
-        return tf.nn.softmax(x)
-
-    def get_params(self):
-        return []
-
-
-class Flatten(Layer):
-
-    def __init__(self):
-        pass
-
-    def set_input_shape(self, shape):
-        self.input_shape = shape
-        output_width = 1
-        for factor in shape[1:]:
-            output_width *= factor
-        self.output_width = output_width
-        self.output_shape = [shape[0], output_width]
-
-    def fprop(self, x):
-        return tf.reshape(x, [-1, self.output_width])
-
-    def get_params(self):
-        return []
-
-
-def make_basic_cnn(nb_filters=64, nb_classes=10,
-                   input_shape=(None, 28, 28, 1)):
-    layers = [Conv2D(nb_filters, (8, 8), (2, 2), "SAME"),
-              ReLU(),
-              Conv2D(nb_filters * 2, (6, 6), (2, 2), "VALID"),
-              ReLU(),
-              Conv2D(nb_filters * 2, (5, 5), (1, 1), "VALID"),
-              ReLU(),
-              Flatten(),
-              Linear(nb_classes),
-              Softmax()]
-
-    model = MLP(layers, input_shape)
-    return model
+class ModelBasicCNN(Model):
```

Comment:
@goodfeli can you confirm you are OK with changing the tutorial model in pure TF to this implementation?

Reply:
since it's the tutorials no one should depend on this code and @david-berthelot says the runtime and accuracy are the same. so yeah, let's go for it

Reply:
SGTM